frontend/src/cljs/imas_seamap/core.cljs

```diff
@@ -141,6 +142,8 @@
     :map.layer/opacity-changed            [mevents/layer-set-opacity]
     :map.layers/filter                    [mevents/map-set-layer-filter]
     :map.layers/others-filter             mevents/map-set-others-layer-filter
+    :map.layer/get-legend                 [mevents/get-layer-legend]
+    :map.layer/got-legend                 mevents/got-layer-legend
```

Comment:
Not suggesting we change at this point, but I wonder if this is the sort of thing that would be better off as a component?  You could use the async programming features, and not have to reframe namespace with all the callback events.  Possibly one to talk through with OG.

Reply:
[This kind of thing](https://stackoverflow.com/a/60472172) could probably be used to avoid polluting the reframe namespace, but we're still left with the side effects issue.